lib/idx/remediate.ts

```diff
@@ -139,13 +144,28 @@ function handleIdxError(e, flow, remediator?) {
   throw e;
 }
 
+function getRelatesToAction(idxResponse: IdxResponse, remediationValues: RemediationValues) {
+  const knownActionParameters = ['resend'];
+
+  const mathcingActionParameter =
+    Object.keys(remediationValues).find(valueName => knownActionParameters.includes(valueName));
+  if (mathcingActionParameter) {
+    const hasRelatedAction = idxResponse.neededToProceed.some(remediation => {
+      return remediation.relatesTo?.value && Object.keys(remediation.relatesTo.value).includes(mathcingActionParameter);
```

Comment:
`resend` object is available under `relatesTo` key:
```
"relatesTo": {
    "type": "object",
    "value": {
      "resend": {
        "rel": [
          "create-form"
        ],
        "name": "resend",
        "href": "https://dev-org.okta.com/idp/idx/challenge/resend",
        "method": "POST",
        "produces": "application/ion+json; okta-version=1.0.0"
 ...      
```

 but it looks like there is no direct link between it and `actions`:
 
 
<img width="326" alt="Screenshot 2021-06-29 at 12 48 19" src="https://user-images.githubusercontent.com/71440851/123790291-cf7bb200-d8e6-11eb-9d1a-810fe2a71688.png">

Reply:
ideally, there should be only one `resend` related action per response. I think we can make the same assumption here, just add a comment about the assumption in code should be fine.

Reply:
I was assuming `relatesTo` points to an action so we could trigger it without additional mapping (e.g. `resend` param -> `needToProceed[i].relatesTo.value['resend'].name` -> `idxResponse.action[resendActionName]`. `relatesTo` check was added to reinforce action matching since current code is only testing the inclusion of `-resend` suffix into action name (i.e.  `'currentAuthenticatorEnrollment-resend'.includes('resend')`. With explicit mapping of params to actions this should not be necessary.

Reply:
right, it would be good to find clue based on the response. My initial concern was the actions field are generated from `idx-js` which owned by our team, we probably can follow or introduce patterns at that step for action discovery instead of looking it up again during remediation. But all those can be future works.